CONTRIBUTING.md

```diff
@@ -32,63 +32,251 @@ Follow [@zitadel](https://twitter.com/zitadel) on twitter
+| backend         | Service that serves the grpc(-web) and RESTful API                 | [go](https://go.dev)                                                        | [API implementation](./internal/api/grpc)          |
+| console         | Frontend the user interacts with after he is logged in             | [Angular](https://angular.io), [Typescript](https://www.typescriptlang.org) | [./console](./console)                             |
+| login           | Server side rendered frontend the user interacts with during login | [go](https://go.dev), [go templates](https://pkg.go.dev/html/template)      | [./internal/api/ui/login](./internal/api/ui/login) |
+| API definitions | Specifications of the API                                          | [Protobuf](https://developers.google.com/protocol-buffers)                  | [./proto/zitadel](./proto/zitadel)                 |
+| docs            | Project documentation made with docusaurus                         | [Docusaurus](https://docusaurus.io/)                                        | [./docs](./docs)                                   |
 
 Please validate and test the code before you contribute.
 
+We add the label "good first issue" for problems we think are a good starting point to contribute to ZITADEL.
+
+- [Issues for first time contributors](https://github.com/zitadel/zitadel/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22)
+- [All issues](https://github.com/zitadel/zitadel/issues)
+
+
 ### Backend / Login
 
-To keep the code clean and understandable we use [golangci-lint](https://golangci-lint.run). We recommend to format the code with this linter while working on ZITADEL to simplify the review process. The configuration is located [here](./.golangci.yaml).
+<!-- TODO: Add links -->
+
+By executing the commands from this section, you run everything you need to develop the ZITADEL backend locally.
+Using Docker Compose, you run a CockroachDB on your local machine.
+With [goreleaser](https://opencollective.com/goreleaser), you build a debuggable ZITADEL binary and run it using [delve](https://github.com/go-delve/delve).
+Then, you test your changes via the console your binary is serving at http://<span because="breaks the link"></span>localhost:8080 and by verifying the database.
+Once you are happy with your changes, you run end-to-end tests and tear everything down.
 
-To start the backend with a debugger run the [`main.go`-file](./main.go) located in the root of ZITADEL and provide the arguments and env-variables from below. Ensure that the database is running by running `docker compose -f ./build/local/docker-compose.yml up db`. For additional information please use the documentation of your IDE.
+The commands in this section are tested against the following software versions:
+<!-- TODO: complete, link and update -->
+- Docker version 20.10.17
+- Goreleaser version v1.8.3
+- Go version 1.17.5
```

Comment:
Development is done with the last stable release. The minimum required version can be found in the go.mod file.